Hdf5 For Windows V2

jpopelar opened the pull request:
Remake of PR #29826 due to branch conflicts and duplicated commits

Commit:
Moving shared and staticmt builds into new WindowsPackage class

Commit:
Formatting

Commit:
sqlite gotten past build stage, working on install

Commit:
sqlite building and installing without readline

Commit:
Saving progress, nmake doesn't work with GNU Makefiles, so will likely need to go down CMake route

Commit:
openblas building and installing with Bilder CMake arguments; will move towards adding logic for other options

Commit:
Following on OpenBLAS issue #2826 where -DDYNAMIC_ARCH doesn't work with MSVC, use -DTARGET=GENERIC for now

Commit:
bzip2 packaged, needs to be built with ~shared since makefile.msc doesn't have a shared build recipe

Commit:
Setting shared default to false, phasing out Unix specific file upkeep

Commit:
Restoring default and conflict as per Greg Becker's suggestion

Commit:
Formatting

Commit:
Saving progress on boost and py-setuptools

Commit:
Fixing library extension on Windows

Commit:
Setting default on bzip2 shared to be false on Windows; avoids spec conflicts for upstream packages

Commit:
Saving progress, looking at a path separator issue for when boostrap.bat is called for

Commit:
Saving progress, go through cmd to invoke bootstrap.bat

Commit:
Boost boostrap complete, working on build

Commit:
Boost building with standard b2 call

Commit:
Style edits

Commit:
hdf5 building serial

Commit:
Reverting WindowsPackage hierarchy in boost, hdf5 building parallel for Windows

Commit:
Using CMake to add MT flags on request, also adding default policy to implement setting

Commit:
Bump CMake min version

Commit:
Formatting

Commit:
Restoring WindowsPackage inheritance

Commit:
Can't build staticmt and shared at the same time

Commit:
Saving progress, SUPRA_SEARCH_PATH makes negative progress

Commit:
Styling

Commit:
Addressing some code review suggestions from PR #29826

Commit:
Updating supported Windows packages on Getting Started docs

Commit:
Removing top level shared build for Windows, also adding conflicts for when staticmt is called for on non-Windows platforms

Commit:
Undoing merge, prepping for an onto rebase with develop

Commit:
Cleaning up some syntax

Commit:
Fix imports for hdf5

Commit:
Revert "Following on OpenBLAS issue #2826 where -DDYNAMIC_ARCH doesn't work with MSVC, use -DTARGET=GENERIC for now"

This reverts commit 0abe560340ebe8e27ce8ea60f3ddf2ad43817fb6.

Commit:
Revert "openblas building and installing with Bilder CMake arguments; will move towards adding logic for other options"

This reverts commit 7cf8a84ee0684ab57b06c12958f880f8e14a43e1.

Commit:
Revert "Saving progress, nmake doesn't work with GNU Makefiles, so will likely need to go down CMake route"

This reverts commit 6e1047627e9b50711b2026025f996c64dbaff825.

Commit:
Revert "sqlite building and installing without readline"

This reverts commit 4d381a57b465fd6b22f3eeb6e4a90f1becd76b25.

Commit:
Revert "Style edits"

This reverts commit be8f826a4210578d92bf58ca9026f217302d2878.

Commit:
Revert "Boost building with standard b2 call"

This reverts commit 671b2077d7950db2bd00ed734fa238c9f1d50e96.

Commit:
Revert "Saving progress, go through cmd to invoke bootstrap.bat"

This reverts commit 81ce3517dad89571f37b22a8bf056d6feb7e5241.

Commit:
Revert "Saving progress, looking at a path separator issue for when boostrap.bat is called for"

This reverts commit 229dd15524326b29327099186a0042f19116c7d7.

Commit:
Styling post revert

Commit:
More feedback from Peter

Commit:
Fixing pcre function

Commit:
Addressing some more feedback with package focus and styling

Commit:
Saving progress, adding draft of msmpi package file

Commit:
Saving progress

Commit:
Styling

Commit:
Adding dummy dlls for directory search tests

Commit:
Adding static libb.lib; search tests now passing

scheibelp commented on var/spack/repos/builtin/packages/sqlite/package.py:
```diff
@@ -48,6 +48,8 @@ class Sqlite(AutotoolsPackage):
     variant('rtree', default=True, description='Build with Rtree module')
 
+    # This isn't ideal but since we don't have a "not" style syntax we have to condition
+    # the depends_on based on the actual platform vs. the spec's platform
```
This looks vestigial (or some accompanying changes did not make it over)

scheibelp commented on var/spack/repos/builtin/packages/perl/package.py:
```diff
@@ -329,8 +328,8 @@ def _setup_dependent_env(self, env, dependent_spec, deptypes):
-        if is_windows:
-            env.append_path('PATH', self.prefix.bin)
+            if is_windows:
+                env.append_path('PATH', self.prefix.bin)
```
Was something going wrong when you didn't make this change?

I ask because I think `prefix.bin` will always be added to PATH (this is performed for build/run dependencies in build/run contexts, respectively), so I'm surprised these lines existed in the first place.

jpopelar replied:
I'll look into it. spack install perl is hanging for me on Windows so there's something more here

scheibelp commented on var/spack/repos/builtin/packages/netlib-lapack/package.py:
```diff
@@ -82,6 +85,8 @@ class NetlibLapack(CMakePackage):
     depends_on('blas', when='+external-blas')
     depends_on('netlib-xblas+fortran+plain_blas', when='+xblas')
     depends_on('python@2.7:', type='test')
+    depends_on('ninja@1.10.0:', when='platform=windows')
```
Note `CMakePackage` has `depends_on(ninja)`, so unless `netlib-lapack` specifically needs a newer `ninja` version than other CMake packages, it shouldn't be necessary to add this.

chuckatkins commented on lib/spack/spack/build_systems/windows_variants.py:
```diff
@@ -0,0 +1,27 @@
+from spack.package import Package
+
+
+class WindowsPackage(Package):
```
Is this class actually handling *windows* options or *msvc* options? I could be misunderstanding this but there seems to be an assumption that windows = msvc which I don't believe is an assumptions spack should be making.

scheibelp commented on lib/spack/llnl/util/symlink.py:
```diff
@@ -54,7 +53,16 @@ def _win32_junction(path, link):
+        # try using a hard link
+        default_args.insert(0, '/H')
+
+    Executable(command)(*default_args)
```
This undoes: https://github.com/spack/spack/commit/88b1bf751d76e010584b251f482cdfb0ba5351a1

Is there a reason for that?

jpopelar replied:
Not that I can see, maybe got missed by the rebase. Should be fixed now

scheibelp commented on var/spack/repos/builtin/packages/bzip2/package.py:
```diff
@@ -125,7 +148,9 @@ def install(self, spec, prefix):
-            symlink('bzip2', 'bunzip2')
-            symlink('bzip2', 'bzcat')
+        # These files won't be in a Windows installation
+        if sys.platform != 'win32':
```
This should also use `platform=windows` like above.

scheibelp commented on var/spack/repos/builtin/packages/hdf5/package.py:
```diff
@@ -379,10 +382,15 @@ def cmake_args(self):
         ]
 
         api = spec.variants['api'].value
+        if sys.platform == 'win32' and self.spec.satisfies('+staticmt'):
```
I'm wondering if `hdf5~shared` and `hdf5+staticmt` are redundant: can you make a static library for `hdf5` on Windows that doesn't use this?

byrnHDF replied:
In this day and age, Windows "mt" should stop being used. Windows 10+ has improved runtime libraries.

jpopelar replied:
IIRC, the "regular" build links nonshared with /MD flags, so there is a reasonable variant to be had here

scheibelp replied:
I don't think there's a `+staticmt` variant anymore, so I think this `if` needs to be refined.

scheibelp commented on var/spack/repos/builtin/packages/zlib/package.py:
```diff
@@ -3,16 +3,16 @@
-class Zlib(Package):
+# Although zlib comes with a configure script, it does not use Autotools
+# The AutotoolsPackage causes zlib to fail to build with PGI
+class Zlib(Package, WindowsPackage):
```
The only thing `WindowsPackage` provides is `staticmt`. Since this package doesn't use that variant here, does it in fact need to inherit `WindowsPackage`?

scheibelp reviewed (changes requested):
Let me know if any of these requests seem to miss the point.

byrnHDF commented on var/spack/repos/builtin/packages/hdf5/package.py:
```diff
@@ -86,15 +88,16 @@ class Hdf5(CMakePackage):
-    depends_on('cmake@3.12:', type='build')
-
-    depends_on('mpi', when='+mpi')
+    if sys.platform != 'win32':
```
There is a MSMPI available for windows. The spack package is next in review.

johnwparent replied:
Draft MSMPI Spack package can be found [here](https://github.com/spack/spack/pull/30228). Updates are upcoming pending support for the WinWDK/SDK packages in Spack (also in progress).

Commit:
Adding msvc_variants.py, which will succeed windows_variants in the sense that Windows is not the sole OS that can use the compiler

Commit:
[EMPTY] Restarting pipeline

scheibelp commented on var/spack/repos/builtin/packages/sqlite/package.py:
```diff
@@ -48,8 +48,9 @@ class Sqlite(AutotoolsPackage):
 
-    depends_on('readline')
-    depends_on('zlib')
+    if not self.spec.satisfies('platform=windows'):
```
This is causing https://github.com/spack/spack/runs/7472966301?check_suite_focus=true

scheibelp replied:
(there is no `self` here)

scheibelp commented on var/spack/repos/builtin/packages/msmpi/package.py:
```diff
@@ -0,0 +1,29 @@
+    def url_for_version(self, version):
+        return 'https://download.microsoft.com/download/A/E/0/AE002626-9D9D-448D-8197-1EA510E297CE/msmpisetup.exe'
+
+    def install(self):
```
This will probably cause problems because Spack will always call `Package.install()` with arguments. The typical signature is `def install(self, spec, prefix):`

johnwparent committed:
refactor win runtime path object

Commit:
Removing windows_variants and hdf5 dependency

Commit:
Fixing illegal call to self in sqlite

Commit:
Normalizing install arguments call in msmpi

jpopelar commented:
Just pushed up a commit to trim out windows/msvc_variants and addressed your feedback @scheibelp. I'll let the tests go again and LMK if you notice anything else out of the ordinary!

scheibelp commented:
https://github.com/spack/spack/issues/31265 looks related to the style failure, if applying the fix described there works, then could you mention that in that issue?

(one thing that's strange: that issue describes how this would come up for non-Ubuntu systems, but I checked the runner for this style check and it appears to be Ubuntu, so I'm not sure how this would come up there - it is the same error message though)

Commit:
Trying spack style bugfix as suggested by #31265

jpopelar commented:
77796a6 does not look like it did it; just applied @alalazo's suggestion to bootstrap.py by adding a `.local` into the `bin_dir` path. Are there any other issues that discuss this topic?

johnwparent committed:
temp update

Commit:
Reversing 77796a6

Commit:
Try using sys instead of spec for the not conditional in sqlite

Commit:
Iterate over concrete_spec.prefix.bin and concrete_spec.prefix.local.bin

johnwparent committed:
broken commit - revert

Commit:
Use sys.platform in bzip2 recipe

johnwparent committed:
Finalize mixin

jpopelar commented:
@scheibelp Pinging you as per your requests. I'll look into the conflicts if you want to examine anything else

Commit:
[EMPTY] Rerunning tests ahead of fresh code review

scheibelp commented on lib/spack/llnl/util/symlink.py:
```diff
@@ -54,8 +51,6 @@ def _win32_junction(path, link):
         path = os.path.join(parent, path)
         path = os.path.abspath(path)
 
-    CreateHardLink(link, path)
```
Removing this line means this function no longer does anything. Is something going wrong when it does this?

scheibelp commented on var/spack/repos/builtin/packages/hdf5/package.py:
```diff
@@ -86,15 +87,16 @@ class Hdf5(CMakePackage):
-
-    depends_on('mpi', when='+mpi')
+    if sys.platform != 'win32':
+        depends_on('mpi', when='+mpi')
```
Since this PR includes an MPI implementation for Windows, can we allow depends_on('mpi', when=+mpi) for Windows as well? If not (i.e. if this PR will be merged without that) then this should add conflicts('+mpi', when='platform=windows'), and this if can also be removed in that case).

jpopelar replied:
I think to be on the safe side I'll make this a conflicts statement and then see how David Alexander's PR goes along. Will be in next commit

scheibelp commented on var/spack/repos/builtin/packages/hdf5/package.py:
```diff
@@ -86,15 +87,16 @@ class Hdf5(CMakePackage):
-    depends_on('szip', when='+szip')
+    # numactl does not currently build on darwin
+    if sys.platform != 'darwin' and sys.platform != 'win32':
+        depends_on('numactl', when='+mpi+fortran')
```
Have you noticed that `hdf5` on Linux requires `numactl`? I just want to make sure since this PR is generally oriented at building `hdf5` on Windows, but if you noticed an additional constraint for `hdf5` on Linux then that could make sense to add here to (although it might be ideal to split that off into another PR).

scheibelp reviewed (commented):
I have a few questions/requests. I'll do a more-thorough review once this is updated with the latest develop.

Commit:
Restoring a deleted dependency

Commit:
Styling

Commit:
Revert "Iterate over concrete_spec.prefix.bin and concrete_spec.prefix.local.bin"

This reverts commit 64f34d1d331290d537fa88ee6b799cb6cea1e682.

Commit:
Fixing missing import

johnwparent committed:
Proper setter

Commit:
Whitespace styling

Commit:
Undo some aggressive array formatting

Commit:
[EMPTY] Rerunning tests; docs failed

Commit:
Whitespace fix

johnwparent committed:
Remove unused imports

johnwparent committed:
Style

johnwparent committed:
black format

Commit:
Copy johnwparent's fix for installer.py

johnwparent committed:
Install test improvement

Commit:
Revert "Copy johnwparent's fix for installer.py"

This reverts commit 67e6270bcc6c2b55c8c8ae418202cb5ff58e2e08.

Commit:
[EMPTY] Test restart

Commit:
[EMPTY] Test restart

scheibelp commented:
Close/reopen to restart tests

Commit:
Adding a conflicts for mpi for now

jpopelar commented:
@scheibelp PTAL when convenient. Windows unit tests are passing but hitting failures with clingo and Mac

johnwparent commented on lib/spack/spack/test/llnl/util/file_list.py:
```diff
@@ -52,6 +65,15 @@ def header_list():
+plat_static_ext = "lib" if is_windows else "a"
+
+
+plat_shared_ext = "dll" if is_windows else ".so"
```
@jpopelar This line is causing (at least some of) your current CI failures. The extra `.` character in the `".so"` designation for `plat_shared_ext` is causing the string comp for the library names to fail.

jpopelar replied:
Just pushed a fix, thanks!

johnwparent replied:
This fix is already present in the RPath PR, #31930 btw. A lot of the changes from that PR seem to be present in this one, so you may want to check out that PR for future issues.

Commit:
Don't add two dots in the so file extension

Commit:
Saving progress on unit-test debugs

Commit:
Implementation additions

Commit:
Test change

Commit:
Styling

Commit:
Updating conftest, fixing a syntax error in hdf5

Commit:
Fixing missing return value in openblas

Commit:
Black styling

johnwparent commented on var/spack/repos/builtin/packages/netlib-lapack/package.py:
```diff
@@ -105,6 +105,7 @@ class NetlibLapack(CMakePackage):
     depends_on("blas", when="+external-blas")
     depends_on("netlib-xblas+fortran+plain_blas", when="+xblas")
     depends_on("python@2.7:", type="test")
+    depends_on("cmake@3.15:")
```
Is this true for all versions of lapack?

Commit:
[EMPTY] Rerunning CI

Commit:
[EMPTY] CI hung, rerunning

jpopelar commented:
@spackbot run pipeline

Commit:
Removing debug output

Commit:
Restoring develop hdf5

Commit:
Floor version of cmake not needed

scheibelp commented:
@spackbot run pipeline

Commit:
Reverting symlink

Commit:
Fixing whitespace

Commit:
Black styling

Commit:
More whitespace

scheibelp commented on pyproject.toml:
```diff
@@ -1,6 +1,6 @@
 [tool.black]
 line-length = 99
-target-version = ['py27', 'py35', 'py36', 'py37', 'py38', 'py39', 'py310']
+target-version = ['py27', 'py35', 'py36', 'py37', 'py38']
```
This should still test Python 3.9/3.10

jpopelar replied:
Sorry, that was me experimenting with my black installation. Reverted now

scheibelp commented on var/spack/repos/builtin/packages/zlib/package.py:
```diff
@@ -3,15 +3,14 @@
 
 
+# Although zlib comes with a configure script, it does not use Autotools
+# The AutotoolsPackage causes zlib to fail to build with PGI
```
IMO this should be reverted (it doesn't look like there were any changes other than moving a comment)

scheibelp commented on var/spack/repos/builtin/packages/sqlite/package.py:
```diff
@@ -202,6 +204,25 @@ def configure_args(self):
+        if self.spec.satisfies("platform=windows"):
+            nmake = Executable("nmake.exe")
+            print(self.configure_flag_args)
+            nmake('CC = "%s"' % os.environ.get("SPACK_CC"), "Makefile.msc")
```
Did you find this was necessary? We should already be setting CC in the environment: it generally points to the Spack compiler wrapper e.g. `<spack-prefix>/lib/spack/env/cc` and `SPACK_CC` points to the actual compiler path  - in that case we wouldn't get any of the extra `-L` flags inserted by Spack.

scheibelp commented on var/spack/repos/builtin/packages/perl/package.py:
```diff
@@ -368,8 +368,8 @@ def _setup_dependent_env(self, env, dependent_spec, deptypes):
-        if is_windows:
-            env.append_path("PATH", self.prefix.bin)
+            if is_windows:
+                env.append_path("PATH", self.prefix.bin)
```
I see this is moving this, but I'm surprised this is needed: we should already be adding `<prefix>/bin/` to the `PATH` automatically. Does the installation on windows work without this?

scheibelp reviewed (changes requested):
This looks great although I have a few suggestions for cleanup. Also, I don't see any changes to hdf5 itself: does `spack install hdf5+mpi` work on Windows without any further configuration? I would have expected we'd have to add a Windows-specific configuration which encouraged choosing msmpi on Windows (I assume other MPI implementations would fail at the moment).

johnwparent commented on var/spack/repos/builtin/packages/msmpi/package.py:
```diff
@@ -0,0 +1,34 @@
+    version(
+        "10.0",
+        sha256="7dae13797627726f67fab9c1d251aec2df9ecd25939984645ec05748bdffd396",
+        extension="exe",
```
Does this work? AFAIK this should fail because we'll still try to extract the extension and fail (because it's a binary). Should this be set to `expand=False`?

johnwparent commented on var/spack/repos/builtin/packages/sqlite/package.py:
```diff
@@ -202,6 +204,25 @@ def configure_args(self):
+
+    def build(self, spec, prefix):
+        if self.spec.satisfies("platform=windows"):
+            nmake = Executable("nmake.exe")
```
This is no longer necessary, `nmake` should be set by the build environment setup.

Commit:
Addressing some johnwparent review points

Commit:
Styling

Commit:
hdf5+mpi now working with msmpi by direct adding of msmpi to packages.yaml

Commit:
Reinstate non-Windows dependencies

scheibelp commented on var/spack/repos/builtin/packages/hdf5/package.py:
```diff
@@ -196,13 +196,16 @@ class Hdf5(CMakePackage):
+    # Further, check for standard mpi in this same Windows-excluding loop
+    for plat in ['cray', 'darwin', 'linux']:
+        depends_on("pkgconfig", when="platform=%s" % plat, type="run")
+        depends_on("mpi", when="+mpi, platform=%s" % plat)
```
You can actually keep `depends_on(mpi, when=+mpi)` and do not need to make `depends_on(mpi)` platform-specific: the idea is that every platform needs to build the MPI virtual with `+mpi`, but Windows requires the additional constraint that you specifically need `msmpi`. 

(this loop is still required though for the pkgconfig dependency)

scheibelp commented on var/spack/repos/builtin/packages/msmpi/package.py:
```diff
@@ -0,0 +1,42 @@
+
+    def install(self, spec, prefix):
+        installer = Executable("msmpisetup.exe")
+        installer("-unattend", "-installroot", prefix)
```
At one point I think you mentioned `-installroot` wasn't working as expected (i.e. the installer was ignoring it and installing to some common location) - is that still true?

Furthermore, with `-unattend`, does this require user interaction when the install fails (e.g. when `msmpi` is already installed)?

jpopelar replied:
For the first point, I mentioned that the installation was empty even with the option. I eventually got around this by declaring msmpi as a non-buildable external in my packages.yaml

Pursuant to the second point, there is no user input required during installation failure, as I've tested

johnwparent replied:
I can confirm behavior on my local setup as well.

scheibelp replied:
> I eventually got around this by declaring msmpi as a non-buildable external in my packages.yaml

I'm not clear how that addresses the issue of being unable to specify the Spack prefix. If I understand: you have confirmed that this does in fact install `msmpi` when it has not been installed yet - did I get that right? If so, the docstring for the package should mention that `msmpi` installs to a predetermined location.

scheibelp commented on var/spack/repos/builtin/packages/sqlite/package.py:
```diff
@@ -4,6 +4,7 @@
+import sys
 from tempfile import NamedTemporaryFile
 
 import spack.platforms
```
It just occurred to me: `sqlite` is an `AutotoolsPackage`, and I thought that class defined a conflict with `platform=windows`. Is `sqlite` needed as a dependency of `hdf5`? If not, then we could move it to another PR. If you want to keep it, I would want to make sure we can install/concretize it. We could leverage multi-build-system support to make it easier to install this on Windows in that case.

jpopelar replied:
I'll be phasing the work on sqlite out in my next commit; stay tuned

johnwparent replied:
Can you repurpose those commits into a separate PR? Sqlite support is something I'm very interested in (and assumed this PR would cover) and if you don't have the cycles to do the extra work, I can/will happily take over/add it to my mega package PR.

scheibelp commented on lib/spack/spack/detection/path.py:
```diff
@@ -234,7 +234,6 @@ def by_executable(packages_to_check, path_hints=None):
                 exe_pattern_to_pkgs[exe].append(pkg)
         # Add Windows specific, package related paths to the search paths
         path_hints.extend(compute_windows_program_path_for_package(pkg))
-
```
This looks vestigial (no modifiers to this file other than whitespace)

jpopelar replied:
Probably leftover damage from some debug output; fixed now

scheibelp reviewed (changes requested):
I few additional requests. Let me know if you disagree with any

Commit:
Styling

Commit:
Black wants double quotes

Commit:
Code review feedback

Commit:
Black styling

Commit:
flake8 styling

jpopelar commented:
While I wait on the GitLab CI, can anyone give me details on bootstrap/macos-gnupg-sources? I'm not familiar with what goes into this set of the tests

scheibelp commented:
> While I wait on the GitLab CI, can anyone give me details on bootstrap/macos-gnupg-sources? I'm not familiar with what goes into this set of the tests

The error `[2022-11-15-17:44:47.766694] Error: Failed to install gnupg due to TimeoutError: The read operation timed out` does not look related to your changes, so I'm inclined to re-run it. To do that I'll need to close and reopen the PR.

Commit:
Adding staticmt variants for all prerequisites that call for one in bilder

Commit:
Adding staticmt to zlib, bumping required cmake across packages with MT builds

Commit:
Formatting

Commit:
Revert "Boost boostrap complete, working on build"

This reverts commit b64c6b44f0443558eeb5669fd96d576389bf9061.

Commit:
Revert "Saving progress on boost and py-setuptools"

This reverts commit 7198d1304e0ed86353de14b1f61958bc5933ed66.

Commit:
Shouldn't be using CMakePackage

Commit:
Styling

Commit:
Addressing some feedback from scheibelp

Commit:
Styling

Commit:
Conforming with black standards

Commit:
More styling

Commit:
Revamp run_after

Commit:
Review update

Commit:
cleanup

Commit:
Refactor library list test

Commit:
style fixup

Commit:
Fixup internal link step

Commit:
Review changes

Commit:
Fixup copy tree for Windows symlinks

Commit:
Restoring to develop after merge confs

Commit:
Addressing scheibelp feedback

Commit:
Styling and post rebase cleanup

Commit:
Another pass of styling and rebase cleanups

jpopelar commented:
@spacbot run pipeline

jpopelar commented:
@spackbot run pipeline

scheibelp commented:
Thanks @jpopelar!

jpopelar commented:
You're welcome @scheibelp, have a good Thanksgiving

skosukhin commented on var/spack/repos/builtin/packages/hdf5/package.py:
```diff
@@ -196,13 +196,16 @@ class Hdf5(CMakePackage):
 
     # The compiler wrappers (h5cc, h5fc, etc.) run 'pkg-config'.
-    depends_on("pkgconfig", type="run")
+    # Skip this on Windows since pkgconfig is autotools
+    for plat in ["cray", "darwin", "linux"]:
+        depends_on("pkgconfig", when="platform=%s" % plat, type="run")
```
I'm late with my comments but nonetheless.
1. I disagree with the comment: `pkgconfig` is not `autotools`.
2. The compiler wrappers will not work without `pkg-config` in the `PATH`.

I understand that the compiler wrappers of `hdf5` a rarely used but if `pkg-config` works on Windows, the code above should be reverted. Otherwise, the wrappers should be removed after the installation to avoid confusion.
